### DISCUSSION TITLE
How to removeAction or removeCommand after adding any addAction or addCommand

### DISCUSSION THREAD

**Discussion by Rajgupta7080** [microsoft/monaco-editor#4302] [Q&A]
`const handleCommandLogic = () => {
   
        const value = filteredOptions.length > 0 ? currentValue?.toLowerCase() : "";
        const lineContent = editor.getModel().getLineContent(editor.getPosition().lineNumber);
     
        if (lineContent.includes("devices: ") || lineContent.includes("groups: ")) {
          handleDropdownDevices(value);
        } else {
          handleDropdownChange(value);
        }
      };
     
     editor.addAction({
        id: 'my-command',
        label: 'My Command',
        keybindings: [monaco.KeyCode.Enter],
        run: handleCommandLogic,
      });`
      
      I want to remove this action how can i do

**Comment by rcjsuen** [comment]
See solution [here](https://microsoft.github.io/monaco-editor/playground.html?source=v0.47.0#XQAAAAKgAgAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscw2IvwRgmSaumWUF-TBOsVJ4S4FwfCDQb5-2uUHXTVAefbVSd9IzaZ_vv9VR9KugsKimARHSWlrTPOQHROITHxmAeFEs9_m3VHdSe7_Etv6WpgzlH4mYHHtn1w7fn-LcvEK49bG3A9p5jOiai9CxwbRq1C_XeYqlR_MCdm3ldTqb044JlL2a3tEYPLKHudje8qmhL4ysHxsG6SEhb0YrzQZ5Gu3vqNJ127wofBCxQGnt942ZdIMKuQpM_DclJgYth5PH1le6v3oU7bbEkOE5UBJ8wf708FA64rlZedW5TNEaFKfP5ZSVC641ZCGGGNlZs-ii1kQwy3Qn7NAGaXMNvkjSrNhIb9bbb5GQdgo6PwjktUbh8UL500ZHL-Yd3XIb_ZhJGCTPvS-PvL2gYbdEWc0jECJx3FpN0b18dTFfPDwusagP4aPleIDmxixyFSrLzAsQHo2uPHH7yTQi4zoZSbOKNgYu1AqA6__319KwA).

`addAction` returns an `IDisposable` so you can calll `dispose()` on the returned object to remove the action when desired.

1. Open the browser console.
2. F1  > temporary action
3. Run it a few times, you should see stuff printed to the console.
4. F1 > dispose action
5. You should see the disposal notification in the console.
6. F1 > temporary action
7. Now you can no longer see the temporary action in the F1 list.

```JavaScript
const value = /* set from `myEditor.getModel()`: */ `function hello() {
	alert('Hello world!');
}
`;

// Hover on each property to see its docs!
const myEditor = monaco.editor.create(document.getElementById("container"), {
	value,
	language: "javascript",
	automaticLayout: true,
});
const disposable = myEditor.addAction({
	id: "monaco-bug",
	label: "temporary action",
	run: () => {
		console.log(`ran ${new Date()}`);
	}
})
myEditor.addAction({
	id: "monaco-bug-cleanup",
	label: "dispose action",
	run: () => {
		console.log("Disposing...");
		disposable.dispose();
		console.log("Disposed...");
	}
})
```
